semantics/language/execution/io-buffered.k

```diff
@@ -102,217 +88,188 @@ module C-IO-BUFFERED
      rule locations(.List) => .Set
-     rule locations(ListItem(bwrite(Loc:SymLoc, _)) L:List)
+     rule locations(ListItem(bwrite(Loc::SymLoc, _)) L::List)
           => SetItem(Loc) locations(L)
 
-     rule <k> readBytes(Loc:SymLoc, Size:Int, T:Type)
+     rule <k> readBytes(Loc::SymLoc, Size::Int, T::Type)
                => checkEffectiveType(T, getEffectiveType(Loc))
                ~> readBytes-aux(Loc, Size, .List, T)
           ...</k>
-          <thread-id> ThreadId:Int </thread-id>
+          <thread-id> ThreadId::Int </thread-id>
           requires notBool ((isThreadDuration(Loc) orBool isAutoDuration(Loc))
                     andBool (ThreadId =/=K getThreadId(Loc)))
           [structural]
      rule <k> (.K => UNSPEC("EIO6", "Attempting to access an object with thread or auto storage duration in a thread other than the one in which the object is associated.", "6.2.4:4-5"))
-               ~> readBytes(Loc:SymLoc, _, _)
+               ~> readBytes(Loc::SymLoc, _, _)
           ...</k>
-          <thread-id> ThreadId:Int </thread-id>
+          <thread-id> ThreadId::Int </thread-id>
           requires (isThreadDuration(Loc) orBool isAutoDuration(Loc))
                andBool (ThreadId =/=K getThreadId(Loc))
           [structural]
 
      // loc, size in bytes, aux list
      syntax KItem ::= "readBytes-aux" "(" SymLoc "," Int "," List "," Type ")"
-     rule readBytes-aux(Loc:SymLoc, Size:Int, Aux:List, T:Type)
-          => readByte(Loc, T)
-          ~> readBytes-aux(Loc +bytes 1, Size -Int 1, Aux, T)
-          requires Size >Int 0
-          [structural]
-     rule readBytes-aux(_, 0, Aux:List, _) => dataList(bvalues(Aux))
+     rule readBytes-aux(_, 0, Aux::List, _) => dataList(Aux)
           [structural]
-     rule (tv(V:CValue, T:Type) => .K)
-          ~> readBytes-aux(_, _,
-               (Aux:List => Aux ListItem(tv(V, T))), _)
-          [structural]
-
-     syntax List ::= bvalues(List) [function]
-     rule bvalues(ListItem(tv(K:K, _)) L:List) => ListItem(K) bvalues(L)
-     rule bvalues(.List) => .List
 
      syntax Error ::= "errorReadOutOfBounds"
                     | "errorUnseqEffects"
                     | "errorNonVolatileLVal"
 
      syntax KItem ::= readByte(SymLoc, Type)
-     rule <k> readByte(loc(Base:SymBase, Offset:Int), T:Type)
+     rule <k> readBytes-aux(loc(Base::SymBase, Offset::Int), Size::Int, Aux::List, T::Type)
                // TODO(chathhorn): do we really need to check this stuff every
                // byte?
-               => assert(notBool (loc(Base, Offset) in Locs),
-                    errorUnseqEffects)
-               ~> assert(Offset <Int Len, errorReadOutOfBounds)
-               ~> assert((notBool (Volatile in getQualsAtOffset(ObjT, Offset)))
-                    orBool isVolatileType(T),
-                    errorNonVolatileLVal)
-               ~> tv(V, t(.Set, no-type))
+               => #if notBool (loc(Base, Offset) in Locs) #then .K #else
```

Comment:
The point is not to have decent performance and period, because gcc has much better performance. The point is to have a semantics which is readable, easily maintainable, and has decent performance. I wasted months using the C semantics for verification as opposed to weeks for Java and JavaScript because of how the semantics is written. Refactoring is already painful enough.

Revert the manual inlining. If you feel so strongly about it, implement a inlining pass.